dev/TabView/TabView.cpp

```diff
@@ -0,0 +1,248 @@
+                        m_indexToSelectOnSelectionChanged = std::optional(index);
+                        break;
+                    }
+
+                    // try the next item
+                    index++;
+                    if (index >= numItems)
+                    {
+                        index = 0;
+                    }
+                } while (index != startIndex);
+            }
+        }
+    }
+
+    UpdateTabWidths();
+
+    __super::OnItemsChanged(item);
+}
+
+void TabView::OnSelectionChanged(const winrt::IInspectable& sender, const winrt::SelectionChangedEventArgs& args)
+{
+    if (m_indexToSelectOnSelectionChanged)
+    {
+        SelectedItem(Items().GetAt(m_indexToSelectOnSelectionChanged.value()));
+        m_indexToSelectOnSelectionChanged = {};
+    }
+
+    UpdateTabContent();
+}
+
+void TabView::UpdateTabContent()
+{
+    if (auto tabContentPresenter = m_tabContentPresenter.get())
+    {
+        if (!SelectedItem())
+        {
+            tabContentPresenter.Content(nullptr);
+            tabContentPresenter.ContentTemplate(nullptr);
+        }
+        else
+        {
+            auto container = ContainerFromItem(SelectedItem()).as<winrt::ListViewItem>();
+            if (container)
+            {
+                tabContentPresenter.Content(container.Content());
+                tabContentPresenter.ContentTemplate(container.ContentTemplate());
+            }
+        }
+    }
+}
+
+void TabView::CloseTab(winrt::TabViewItem const& container)
+{
+    if (auto item = ItemFromContainer(container))
+    {
+        uint32_t index = 0;
+        if (Items().IndexOf(item, index))
+        {
+            auto args = winrt::make_self<TabViewTabClosingEventArgs>(item);
+
+            m_tabClosingEventSource(*this, *args);
+
+            if (!args->Cancel())
+            {
+                Items().RemoveAt(index);
+            }
+        }
+    }
+}
+
+void TabView::OnScrollDecreaseClick(const winrt::IInspectable& sender, const winrt::RoutedEventArgs& args)
+{
+    if (auto scrollViewer = m_scrollViewer.get())
+    {
+        scrollViewer.ChangeView(std::max(0.0, scrollViewer.HorizontalOffset() - c_scrollAmount), nullptr, nullptr);
+    }
+}
+
+void TabView::OnScrollIncreaseClick(const winrt::IInspectable& sender, const winrt::RoutedEventArgs& args)
+{
+    if (auto scrollViewer = m_scrollViewer.get())
+    {
+        scrollViewer.ChangeView(std::min(scrollViewer.ScrollableWidth(), scrollViewer.HorizontalOffset() + c_scrollAmount), nullptr, nullptr);
+    }
+}
+
+void TabView::UpdateTabWidths()
+{
+    double tabWidth = DoubleUtil::NaN;
+
+    if (TabWidthMode() == winrt::TabViewWidthMode::Fixed)
+    {
+        // Tabs should all be the same size, proportional to the amount of space.
+        double minTabWidth = unbox_value<double>(SharedHelpers::FindResource(c_tabViewItemMinWidthName, winrt::Application::Current().Resources(), box_value(c_tabMinimumWidth)));
```

Comment:
>unbox_value<double>(SharedHelpers::FindResource(c_tabViewItemMinWidthName, winrt::Application::Current().Resources(), box_value(c_tabMinimumWidth)) [](start = 29, length = 147)

If you use template, we may simplify the usage to:
auto minTabWidth = SharedHelpers::FindResource<double>(c_tabViewItemMinWidthName, winrt::Application::Current().Resources(), c_tabMinimumWidth)

You don't need to box, then unbox value. and ValueHelper could help to Box anything if necessary.